index.js

```diff
@@ -1112,10 +1109,26 @@ module.exports = class Autobase extends ReadyResource {
     await this._makeLinearizer(this.system)
   }
 
-  async initialFastForward (key, length, timeout) {
+  async initialFastForward (key, timeout) {
     const encryptionKey = this._viewStore.getBlockKey(this._viewStore.getSystemCore().name)
 
     const core = this.store.get({ key, encryptionKey, isBlockKey: true })
+
+    // get length from network
+    const length = await new Promise((resolve, reject) => {
+      const timer = setTimeout(() => {
+        core.off('append', resolve)
```

Comment:
you need to `off` the function passed to append below, so save that in a variable first